src/Joe/mappings/handlerFarm.ts

```diff
@@ -0,0 +1,106 @@
+import { Address, BigInt, Bytes, log } from "@graphprotocol/graph-ts";
+import { JoeFarmData } from "../../../generated/schema";
+import { JoeMasterChefJoeV3 as MasterChefJoeV3Contract } from "../../../generated/JoeMasterChefJoeV3/JoeMasterChefJoeV3";
+import { JoeMasterChefJoeV2 as MasterChefJoeV2Contract } from "../../../generated/JoeMasterChefJoeV2/JoeMasterChefJoeV2";
+
+// Is the same abi for both versions
+import { JoeSimpleRewarderPerSec as RewarderContract } from "../../../generated/JoeMasterChefJoeV2/JoeSimpleRewarderPerSec";
+import { convertBINumToDesiredDecimals, convertBytesToAddress } from "../../utils/converters";
+import { JoeMasterChefV2Address, JoeMasterChefV3Address, ZERO_ADDRESS, ZERO_BD, ZERO_BI } from "../../utils/constants";
+
+export function handlePool(
+  txnHash: Bytes,
+  blockNumber: BigInt,
+  timestamp: BigInt,
+  poolId: BigInt,
+  eventType: string,
+  version: string,
+  lpToken: Address,
+): void {
+  let entity = JoeFarmData.load(txnHash.toHex());
+  if (!entity) entity = new JoeFarmData(txnHash.toHex());
+
+  entity.blockNumber = blockNumber;
+  entity.blockTimestamp = timestamp;
+  entity.poolId = poolId;
+
+  entity.event = eventType;
+  entity.version = version;
+
+  let poolAllocPoints = ZERO_BI;
+  let poolJoePerSec = ZERO_BD;
+  let totalJoePerSec = ZERO_BD;
+  let totalAllocPoints = ZERO_BI;
+  let poolLastRewardTimestamp = ZERO_BI;
+  let poolAccJoePerShare = ZERO_BD;
+  let poolLpToken = ZERO_ADDRESS;
+  let rewarderAddress = ZERO_ADDRESS;
+
+  if (version == "2") {
+    let contract = MasterChefJoeV2Contract.bind(JoeMasterChefV2Address);
+    let totalJoePerSecResult = contract.try_joePerSec();
+
+    if (totalJoePerSecResult.reverted) {
```

Comment:
Please log some details (or use ! reverted to avoid the empty if block).
(applies to all places like this)